staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -185,7 +185,10 @@ func translateServicePortToTargetPort(ports []string, svc corev1.Service, pod co
 			return nil, err
 		}
 
-		if int32(portnum) != containerPort || localPort == "" {
+		// should fail when localPort is empty (=> use random local port)
+		localportnum, err := strconv.Atoi(localPort)
+
+		if int32(portnum) != containerPort || localPort == "" || (int32(localportnum) != containerPort && err == nil) {
 			translated = append(translated, fmt.Sprintf("%s:%d", localPort, containerPort))
 		} else {
 			translated = append(translated, fmt.Sprintf("%d", containerPort))
```

Comment:
```suggestion
		// convert the resolved target port back to a string
		remotePort = strconv.Itoa(int(containerPort))

		if localPort != remotePort {
			translated = append(translated, fmt.Sprintf("%s:%s", localPort, remotePort))
		} else {
			translated = append(translated, remotePort)
```

Reply:
I would suggest doing something more like this... once we've resolved the service and container ports, convert both to strings, and include both in the output if they differ, and just include one in the output if they are identical. That covers the empty string local port case, the differing numeric cases, and the resolved name case.

Reply:
That change passes all existing unit tests and the unit test added in this PR (which is a good test, thanks for adding that)